### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by tdanylchuk**

### CODE REVIEW

**Review by vasilievip — CHANGES_REQUESTED** [review]
split by commits

File: sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            log.debug("Message doesn't contain span data, new one was created - {}.", newSpan);
+            return newSpan;
+        }
+    }
+
+    private String getSpanName(Endpoint endpoint) {
+        return SpanNameUtil.shorten(MESSAGE_COMPONENT + "::" + endpoint.getEndpointKey());
+    }
+
+    private Span buildSpan(SpanTextMap carrier) {
```
Comment (spolyakov) [inline sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java:88]: Optional<Span>?

File: sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        try {
+            return this.spanExtractor.joinTrace(carrier);
+        } catch (Exception e) {
+            log.error("Exception occurred while trying to extract span from carrier.", e);
+            return null;
+        }
+    }
+
+    @Override
+    public boolean isEnabled(EventObject event) {
```
Comment (spolyakov) [inline sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java:98]: isExchangeCreated()

**Comment by tdanylchuk** [comment]
in favour of https://github.com/Playtika/sleuth-camel/pull/2

**Pull request closed without merging**